src/player.ts

```diff
@@ -1716,6 +1701,46 @@ export default class Player extends FakeEventTarget {
     return this._engine.getDrmInfo();
   }
 
+  private _getSubtitleStyleSheet(): CSSStyleSheet {
+    let element = Utils.Dom.getElementBySelector(`.${this._playerId}.${SUBTITLES_STYLE_CLASS_NAME}`);
+    if (!element) {
+      element = Utils.Dom.createElement('style');
+      Utils.Dom.addClassName(element, this._playerId);
+      Utils.Dom.addClassName(element, SUBTITLES_STYLE_CLASS_NAME);
+      Utils.Dom.appendChild(document.head, element);
+    }
+    return element.sheet;
+  }
+
+  private _resetCustomSubtitleStyles(): void {
+    const sheet = this._getSubtitleStyleSheet();
+    while (sheet.cssRules.length) {
+      sheet.deleteRule(0);
+    }
+  }
+
+  private _applyCustomSubtitleStyles(): void {
+    try {
+      const sheet = this._getSubtitleStyleSheet();
+
+      if (this._config.text.useNativeTextTrack && !this._config.text.useShakaTextTrackDisplay) {
+        sheet.insertRule(`#${this._playerId} video.${ENGINE_CLASS_NAME}::-webkit-media-text-track-display { text-align: ${this._textStyle.textAlign}!important; }`, 0);
+        sheet.insertRule(`#${this._playerId} video.${ENGINE_CLASS_NAME}::cue { ${this._textStyle.toCSS()} }`, 0);
+      } else if (this._config.text.useShakaTextTrackDisplay) {
+        this._resetCustomSubtitleStyles();
+        const flexAlignment = {
+          left: 'flex-start',
+          center: 'center',
+          right: 'flex-end',
+        }
+        sheet.insertRule(`#${this._playerId} .shaka-text-container { padding: 0px 16px; align-items: ${flexAlignment[this._textStyle.textAlign]}!important; }`, 0);
```

Comment:
i don't like how this couples the playkit (which shouldn't even know about shaka existing) to dash adapter and shaka :/
maybe we can add some kind of getter to the base adapter to return the text element or the name of the text element ?